### DISCUSSION TITLE
Current Implementation of Level Set Fire Spread (mode 4)

### DISCUSSION THREAD

**Discussion by FGI21-WF** [firemodels/fds#13293] [General]
Hi All, 
I am doing a bit of digging into the current implementation of the coupled level set fire spread algorithm with the Rothermel model (mode 4 with elliptical spread) and was hoping to get a bit of clarity on some of the things within the code. 
1)	In the LEVEL_SET_FIRESPREAD subroutine, when scanning the cells vertically to find the velocity values at a height of at least 6.1m it is the cell centres that are used. Assuming that the first grid cell is above the vegetation, I would conceptually have thought that the vertical distance would have been determined as the difference between the cell centre of the cells above the first grid cell to the bottom of the first grid cell above the ground. In other words, ‘(ZC(KDUM)-Z(K_LS(IIG,JJG) -1)>=6.1_EB)’ instead of ‘(ZC(KDUM)-ZC(K_LS(IIG,JJG))>=6.1_EB)’. 

      As an example, if the cell heights are 3m, the current implementation will use the fourth vertical cell for subsequent calculations. If the distance is taken to the bottom of the first cell above ground it will be the third vertical cell used in subsequent calculations.   

     I do note that there is a drag force calculation in velo.f90 for level set that attempts to model the influence of the vegetation on the first grid cell for level set. Does this assume that the vegetation is in fact within the first grid cell as opposed to below it or is this just a simple drag model to replicate the effects? If the vegetation is within the cell shouldn’t the vertical scan, consider the height of the vegetation (VEG_HT + 6.1_EB). 

     Chaparral in the standard fuel model in the user guide has a height of 1.83m might be heavily influenced by this approach. This likely will not have much of an impact (if any) on very coarse cells but may impact simulated results with cell heights less than 5m. 

2)	There is a comment in the same subroutine about converting UMF_MAG from m/s to m/min for use in the elliptical model. I am a little confused by this comment as in the LEVEL_SET_SPREAD_RATE subroutine this is converted back to m/s for subsequent use in the LB calculation. Is the conversion to m/min only required for calculating ROTH_FACTOR and subsequent effective midflame windspeed if slope is present? 

3)	 There is an alternative method of progression from the elliptical method called McArthur spread model in the code.  Can a reference for this method of progression be provided?

Many thanks for the help.

**Comment by ericvmueller** [comment]
To your first two points ...and others can correct me

1.  
The aim is to find the velocity at 6.1m above the ground, not above the vegetation. The ground is taken as the solid boundary within cell IIG,JJG, and the surface vegetation is assumed to be _within_ that cell rather than below it. The drag model represents the effect of vegetation in this cell. 

If the ground is defined by solid surfaces at grid planes (i.e. `OBST`) then yes I think your correction is more accurate. However, in the case of complex topography (i.e. `GEOM`) neither Z or ZC can be assumed to be the reference elevation of the ground, so we probably want to use Z_LS in all cases. 

All that being said... we aren't interpolating velocities to an exact height of 6.1m ...or even finding the closest vector to that height, just the first one at or above. So no matter what this is just an approximation. We may want to consider such an interpolation.

2. 
Yes the Rothermel spread equation takes velocity in m/min while the length-to-breadth ratio uses m/s. The alternative would be to adjust the coefficient in the equation but the result should be the same.
